src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -627,15 +664,16 @@ def load_status_file_components(self, initial_load=False):
                         self.__configure_patching_errors = errors['details']
                         self.__configure_patching_top_level_error_count = self.__get_total_error_count_from_prev_status(errors['message'])
 
-    def __get_substatus_message(self, status_file_data, index):
-        return json.loads(status_file_data['status']['substatus'][index]['formattedMessage']['message'])
+    def __get_substatus_message(self, status_file_data, substatus_index):
+        """ Get the substatus payload message by index """
+        return json.loads(status_file_data['status']['substatus'][substatus_index]['formattedMessage']['message'])
```

Comment:
Marking all changes not related to this PR goal. See my other comment on what can be done with these

Reply:
its' counterproductive to revert this change, this logic is very small and straightforward, so I refactor the logic into a function and reuse it in truncation to avoid replication.

Reply:
Miscellaneous changes have the side-effect of making it harder to merge from master. Just a general note @feng-j678. I'm closing this thread to look at the bigger picture.